src/partition/repart.c

```diff
@@ -3640,6 +3640,9 @@ static int partition_target_sync(Context *context, Partition *p, PartitionTarget
         } else if (t->fd >= 0) {
                 struct stat st;
 
+                if (fsync(t->fd) < 0)
+                        return log_error_errno(errno, "Failed to sync changes to '%s': %m", t->path);
+
```

Comment:
this is snake oil, no? i mean, we really don't care about this temporary file, so why sync it?

it makes sense to sync after the copy_bytes() but that's already done via the COPY_FSYNC flag

Reply:
Dropped